include/secp256k1_musig.h

```diff
@@ -375,26 +171,215 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
+ *
+ *  1. Always provide a unique session_id32. It is a "number used once".
+ *  2. If you already know the signing key, message or aggregate public key
+ *     cache, they can be optionally provided to derive the nonce and increase
+ *     misuse-resistance. The extra_input32 argument can be used to provide
+ *     additional data that does not repeat in normal scenarios, such as the
+ *     current time.
+ *  3. If you do not provide a seckey, session_id32 _must_ be UNIFORMLY RANDOM.
+ *     If you do provide a seckey, session_id32 can instead be a counter (that
+ *     must never repeat!). However, it is recommended to always choose
+ *     session_id32 uniformly at random. Note that using the same seckey for
+ *     multiple MuSig sessions is fine.
+ *  4. Avoid copying (or serializing) the secnonce. This reduces the possibility
+ *     that it is used more than once for signing.
+ *
+ *  Remember that nonce reuse will immediately leak the secret key!
+ *
+ *  Returns: 0 if the arguments are invalid and 1 otherwise
+ *  Args:         ctx: pointer to a context object, initialized for signing
+ *  Out:     secnonce: pointer to a structure to store the secret nonce
+ *           pubnonce: pointer to a structure to store the public nonce
+ *  In:  session_id32: a 32-byte session_id32 as explained above. Must be
+ *                     uniformly random unless you really know what you are
+ *                     doing.
+ *             seckey: the 32-byte secret key that will be used for signing if
+ *                     already known (can be NULL)
+ *              msg32: the 32-byte message that will be signed if already known
+ *                     (can be NULL)
+ *       keyagg_cache: pointer to the keyagg_cache that was used to create the aggregate
+ *                     (and tweaked) public key if already known (can be NULL)
+ *      extra_input32: an optional 32-byte array that is input to the nonce
+ *                     derivation function (can be NULL)
+ */
+SECP256K1_API int secp256k1_musig_nonce_gen(
+    const secp256k1_context* ctx,
+    secp256k1_musig_secnonce *secnonce,
+    secp256k1_musig_pubnonce *pubnonce,
+    const unsigned char *session_id32,
+    const unsigned char *seckey,
+    const unsigned char *msg32,
+    const secp256k1_musig_keyagg_cache *keyagg_cache,
+    const unsigned char *extra_input32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
+
+/** Aggregates the nonces of every signer into a single nonce
+ *
+ *  This can be done by an untrusted third party to reduce the communication
+ *  between signers. Instead of everyone sending nonces to everyone else, there
+ *  can be one party receiving all nonces, aggregating the nonces with this
+ *  function and then sending only the aggregate nonce back to the signers.
+ *
+ *  Returns: 0 if the arguments are invalid or if all signers sent invalid
+ *           pubnonces, 1 otherwise
```

Comment:
As @real-or-random noticed this is not true at all. One malicious signer who sees the other signers' nonces and sets their nonce to be the negation of the other signers nonces is sufficient to make this return 0. In that case you wouldn't know which signer cheated. Hence, we need to actually return 1 in that case and set the aggnonce output to the point at infinity. Then we'll notice the malicious signer(s) later, because they can't produce a valid partial signature.